plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1146,6 +1149,47 @@ public void renameTable(ConnectorSession session, ConnectorTableHandle tableHand
         catalog.renameTable(session, ((IcebergTableHandle) tableHandle).getSchemaTableName(), newTable);
     }
 
+    @Override
+    public void setTableProperties(ConnectorSession session, ConnectorTableHandle tableHandle, Map<String, Optional<Object>> properties)
+    {
+        IcebergTableHandle table = (IcebergTableHandle) tableHandle;
+        BaseTable icebergTable = (BaseTable) catalog.loadTable(session, table.getSchemaTableName());
+
+        transaction = icebergTable.newTransaction();
+        UpdateProperties updateProperties = transaction.updateProperties();
+
+        for (Map.Entry<String, Optional<Object>> propertyEntry : properties.entrySet()) {
+            String trinoPropertyName = propertyEntry.getKey();
+            Optional<Object> propertyValue = propertyEntry.getValue();
+
+            switch (trinoPropertyName) {
+                case FILE_FORMAT_PROPERTY:
+                    updateProperty(updateProperties, DEFAULT_FILE_FORMAT, propertyValue, format -> ((IcebergFileFormat) format).toIceberg().toString());
+                    break;
+                case FORMAT_VERSION_PROPERTY:
+                    updateProperty(updateProperties, FORMAT_VERSION, propertyValue, formatVersion -> Integer.toString((int) formatVersion));
+                    break;
+                case PARTITIONING_PROPERTY:
```

Comment:
We should allow changing partitioning, let's have a TODO+issue

Reply:
https://github.com/trinodb/trino/issues/12174